impeller/renderer/backend/vulkan/command_pool_vk.cc

```diff
@@ -14,9 +14,11 @@
 
 namespace impeller {
 
-using CommandPoolMap =
-    std::map<const ContextVK*, std::shared_ptr<CommandPoolVK>>;
+using CommandPoolMap = std::map<uint64_t, std::shared_ptr<CommandPoolVK>>;
 
+// TODO(tbd): This is storing tons of CommandPoolVK's in the test runner since
+//            many contexts are created on the same thread. We need to come up
+//            with a different way to clean these up.
```

Comment:
One option that I'm not 100% sure would work - make the key to this a `std::weak_ptr<ContextVk>`, and whenever the call is made to `GetThreadLocal` iterate through the map and delete dead weak_ptr keys.

Reply:
It should work if we change `ContextVK` to `std::enable_shared_from_this` and update `ContextVK::Create` to return `shared_from_this()` instead of creating a shared pointer, and then call `GetThreadLocal(std::weak_from_this())`.

Reply:
As noted above, this is how it was designed to work (the global variable keeps the pool alive long enough for the worker thread to use it, then destroys it when the thread is killed).  I think we'll probably need to fix this design at some point and is a good reminder why the C++ style guide disallows nontrivially deleted global variables.

The alternative design would be something like make this variable a raw pointer and have the context flush and synchronize the worker thread before deleting the pool.

I'm just removing the crash and keeping the same design for now.